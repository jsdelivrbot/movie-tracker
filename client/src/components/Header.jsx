import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import AuthButton from './AuthButton.jsx';

const Header = (props) => (
  <nav className="navbar navbar-light">
    <ul className="nav navbar-nav">
      <li className="nav-item">
        <Link to="/">Home</Link>
      </li>
      <li className="nav-item">
        <Link to="/resources">Resources</Link>
      </li>
      <li className="nav-item">
        <AuthButton {...props}/>
      </li>
    </ul>
  </nav>
);

export default Header;
