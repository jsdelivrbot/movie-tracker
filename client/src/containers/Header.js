import { connect } from 'react-redux';

import { toggleAuth } from '../redux/auth.js';
import Header from '../components/Header.jsx';

const mapStateToProps = (state) => ({
  isAuthenticated: state.isAuthenticated
});

export default connect(mapStateToProps, { toggleAuth })(Header);
