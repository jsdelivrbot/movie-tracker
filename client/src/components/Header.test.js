import React from 'react';
import { mount } from 'enzyme';
import { mountComponent } from '../utils/testUtils.js';

import Header from './Header.jsx';

test('Header renders correctly', () => {
  const wrapper = mountComponent(Header);
  expect(wrapper).toMatchSnapshot();
});
