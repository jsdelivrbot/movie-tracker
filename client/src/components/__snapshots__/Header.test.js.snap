// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Header>
        <nav
          className="navbar navbar-light"
        >
          <ul
            className="nav navbar-nav"
          >
            <li
              className="nav-item"
            >
              <Link
                replace={false}
                to="/"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Home
                </a>
              </Link>
            </li>
            <li
              className="nav-item"
            >
              <Link
                replace={false}
                to="/resources"
              >
                <a
                  href="/resources"
                  onClick={[Function]}
                >
                  Resources
                </a>
              </Link>
            </li>
            <li
              className="nav-item"
            >
              <AuthButton>
                <button
                  data-test="auth-btn"
                >
                  Sign In
                </button>
              </AuthButton>
            </li>
          </ul>
        </nav>
      </Header>
    </Router>
  </BrowserRouter>
</Provider>
`;
